src/Passes/Source/Commandline/ParameterParser.hpp

```diff
@@ -0,0 +1,105 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+namespace microsoft
```

Comment:
FYI: `qsharp-runtime` repo uses the capitalized namespace names - `using namespace Microsoft::Quantum`. You may want to consider the same.

Reply:
If there is a preference amongst people to use capitalised namespaces, we can change the style. However, the main reason for the `snake_case` namespaces is that increases readability as it allows you to distinguish between `ClassNames` and `namespace_names`. That is, `namespace::ClassName` is different from `Class1::Class2`.

That said, I am happy to change it, so let me know if you prefer that.

Reply:
Up to you. 

Feel free to ignore any my comment marked as "FYI", "Nit". 
If I see anything important then I mark it as IMPORTANT or CRITICAL (so far I haven't seen any serious problems in your code that we need to resolve before merging this PR).
If not marked then feel free to also treat it as "Up to you".